libs/environment/src/hooks/use-environment.spec.tsx

```diff
@@ -0,0 +1,200 @@
+  // @ts-ignore typscript doesn't recognise the mock implementation
+  global.fetch.mockReset();
+  // @ts-ignore typscript doesn't recognise the mock implementation
+  global.fetch.mockImplementation(mockFetch);
+
+  window.localStorage.clear();
+
+  process.env['NX_VEGA_URL'] = mockEnvironmentState.VEGA_URL;
+  process.env['NX_VEGA_ENV'] = mockEnvironmentState.VEGA_ENV;
+  process.env['NX_VEGA_CONFIG_URL'] = mockEnvironmentState.VEGA_CONFIG_URL;
+  process.env['NX_ETHEREUM_PROVIDER_URL'] =
+    mockEnvironmentState.ETHEREUM_PROVIDER_URL;
+  process.env['NX_ETHERSCAN_URL'] = mockEnvironmentState.ETHERSCAN_URL;
+  process.env['NX_VEGA_NETWORKS'] = JSON.stringify(
+    mockEnvironmentState.VEGA_NETWORKS
+  );
+});
+
+afterAll(() => {
+  // @ts-ignore: typescript doesn't recognise the mocked fetch instance
+  fetch.mockRestore();
+  window.localStorage.clear();
+
+  delete process.env['NX_VEGA_URL'];
+  delete process.env['NX_VEGA_ENV'];
+  delete process.env['NX_VEGA_CONFIG_URL'];
+  delete process.env['NX_ETHEREUM_PROVIDER_URL'];
+  delete process.env['NX_ETHERSCAN_URL'];
+  delete process.env['NX_VEGA_NETWORKS'];
+});
+
+describe('useEnvironment hook', () => {
+  it('transforms and exposes values from the environment', () => {
+    const { result } = renderHook(() => useEnvironment(), {
+      wrapper: MockWrapper,
+    });
+    expect(result.error).toBe(undefined);
+    expect(result.current).toEqual(mockEnvironmentState);
+  });
+
+  it('allows for the VEGA_CONFIG_URL to be missing when there is a VEGA_URL present', () => {
+    delete process.env['NX_VEGA_CONFIG_URL'];
+    const { result } = renderHook(() => useEnvironment(), {
+      wrapper: MockWrapper,
+    });
+    expect(result.error).toBe(undefined);
+    expect(result.current).toEqual({
+      ...mockEnvironmentState,
+      VEGA_CONFIG_URL: undefined,
+    });
+  });
+
+  it('allows for the VEGA_URL to be missing when there is a VEGA_CONFIG_URL present', async () => {
+    delete process.env['NX_VEGA_URL'];
+    const { result, waitForNextUpdate } = renderHook(() => useEnvironment(), {
+      wrapper: MockWrapper,
+    });
+    await waitForNextUpdate();
+
+    expect(result.error).toBe(undefined);
+    expect(result.current).toEqual({
+      ...mockEnvironmentState,
+      VEGA_URL: MOCK_HOST,
+    });
+  });
+
+  it('allows for the VEGA_NETWORKS to be missing from the environment', () => {
+    delete process.env['NX_VEGA_NETWORKS'];
+    const { result } = renderHook(() => useEnvironment(), {
+      wrapper: MockWrapper,
+    });
+    expect(result.error).toBe(undefined);
+    expect(result.current).toEqual({
+      ...mockEnvironmentState,
+      VEGA_NETWORKS: {},
+    });
+  });
+
+  it('throws a validation error when NX_VEGA_ENV is not found in the environment', async () => {
+    delete process.env['NX_VEGA_ENV'];
+    const { result } = renderHook(() => useEnvironment(), {
+      wrapper: MockWrapper,
+    });
+    expect(result.error).not.toBe(undefined);
```

Comment:
Chore (blocking?): IMO it'd be nice to expect more detail from the error than simply knowing one occurred.

Reply:
Also on lines 137, 163 and 172

Reply:
the reason i did it this way is that the error messages are generated by a third party, we don't really want to have our test failing because they changed an error text, for example. I was thinking if I create enough test cases here, it will be clear enough.

Reply:
ended up changing this, since the third party got removed, updated the tests to check actual error messages